### PR TITLE
Some addition/fix to profunctors and bifunctors

### PR DESCRIPTION
**Pull request by gr-im**
This PR fix some typos, relax some constraints and add some utilitaries modules (mainly for dealing with Kleisli and Cokleisli Arrow. Because having `Profunctor.From_functor_via_clown` (etc) seems anoying.

### CODE REVIEW

**Comment by xvw** [comment]
Localisation of kind expansion into modules (Kleisli and Cokleisli) seems a good idea. 
Thanks!

**Review by xvw — DISMISSED** [review]

**Review by xvw — DISMISSED** [review]

**Review by xhtmlboi — DISMISSED** [review]

**Review by d-plaindoux — APPROVED** [review]

File: lib/preface_make/cokleisli.mli [hunk 1]
```diff
@@ -0,0 +1,51 @@
+(** [Cokleisli] uses the Cokleisli category to describe arity 2 constructions
+    for arrity 1 constructions, usually using the form:
```
[the remark this answers is not in the document]
Comment (xvw) [reply lib/preface_make/cokleisli.mli]: Done (and also in `Kleisli`, after writing Arrow, Arrow, Arrow, when a word start with `ar` I want add a following `r`. Thanks for your careful review!

[the remark this answers is not in the document]
Comment (gr-im) [reply lib/preface_make/cokleisli.mli]: ???

[the remark this answers is not in the document]
Comment (gr-im) [reply lib/preface_make/cokleisli.mli]: (I don't think "PRs" are the perfect place to make fun of people...

**Review by xvw — APPROVED** [review]

**Pull request merged**